src/plugins/stake-plugins/thorchainSavers/tcSaversPlugin.tsx

```diff
@@ -1087,6 +1140,220 @@ const unstakeRequestInner = async (opts: EdgeGuiPluginOptions, request: ChangeQu
   }
 }
 
+const claimRequest = async (opts: EdgeGuiPluginOptions, request: ChangeQuoteRequest): Promise<ChangeQuote> => {
+  const { wallet, account } = request
+  const { currencyCode, pluginId } = wallet.currencyInfo
+  const nativeAmount = DUST_THRESHOLDS[currencyCode]
+
+  const tokenId = getWalletTokenId(wallet, currencyCode)
+  const isEvm = EVM_PLUGINIDS[pluginId]
+
+  const walletBalance = wallet.balanceMap.get(tokenId) ?? '0'
+
+  if (lt(walletBalance, nativeAmount)) {
+    throw new InsufficientFundsError({ tokenId })
+  }
+
+  await updateInboundAddresses(opts)
+
+  const { primaryAddress, addressBalance } = await getPrimaryAddress(account, wallet, currencyCode)
+
+  const asset = edgeToTcAsset(wallet.currencyConfig, currencyCode)
+  const [chain] = asset.split('.')
+  const poolAddress = inboundAddresses?.find(ia => ia.chain === chain)?.address
+  if (poolAddress == null) {
+    throw new Error('Missing pool address')
+  }
+
+  const utxoSourceAddress = primaryAddress
+  const forceChangeAddress = primaryAddress
+
+  let thorchainWallet = Object.values(account.currencyWallets).find(wallet => wallet.currencyInfo.pluginId === 'thorchainrune')
+
+  if (thorchainWallet == null) {
+    // Make sure the user has a Thorchain wallet to receive the TCY
+    console.log('Thorchain wallet not found, creating one')
+    thorchainWallet = await account.createCurrencyWallet('wallet:thorchainrune', {
+      fiatCurrencyCode: wallet.fiatCurrencyCode,
+      name: lstrings.string_first_thorchainrune_wallet_name,
+      enabledTokenIds: ['tcy']
+    })
+  }
+
+  if (!thorchainWallet.enabledTokenIds.includes('tcy')) {
+    await thorchainWallet.changeEnabledTokenIds([...thorchainWallet.enabledTokenIds, 'tcy'])
+  }
+  const thorchainAddresses = await thorchainWallet.getAddresses({ tokenId: null })
+  const thorchainAddress = thorchainAddresses[0].publicAddress
+
+  let memoValue = `tcy:${thorchainAddress}`
+  let memoType: EdgeMemo['type'] = 'text'
+  if (isEvm) {
+    memoType = 'hex'
+    memoValue = base16.stringify(Buffer.from(memoValue, 'utf8'))
+  }
+
+  // Try to spend right out of the primaryAddress
+  const spendInfo: EdgeSpendInfo = {
+    tokenId,
+    spendTargets: [
+      {
+        publicAddress: poolAddress,
+        nativeAmount: nativeAmount
+      }
+    ],
+    memos: [
+      {
+        type: memoType,
+        value: memoValue
+      }
+    ],
+
+    // Use otherParams to meet Thorchain Savers requirements
+    // 1. Sort the outputs by how they are sent to makeSpend making the target output the 1st, change 2nd
+    // 2. Only use UTXOs from the primary address (index 0)
+    // 3. Force change to go to the primary address
+    otherParams: {
+      enableRbf: false,
+      outputSort: 'targets',
+      utxoSourceAddress,
+      forceChangeAddress
+    },
+    assetAction: { assetActionType: 'claim' },
```

Comment:
Should be
`assetAction: { assetActionType: 'claimOrder' },`
since this is a transaction that is creating a claim order, not the claim itself. The claim itself is the transaction that actually receives the funds.

Reply:
There isn't a second transaction that delivers the funds. Maybe the best option is actually 'stake' since the funds are moved from tcy pool into a staked state?